src/Integration/Binding/CSharpVBProjectBinder.cs

```diff
@@ -87,25 +94,55 @@ public bool IsBindingRequired(BindingConfiguration binding, Project project)
 
         private bool IsFullyBoundProject(BindingConfiguration binding, Project project, Core.Language language)
         {
-            var solutionRuleset = GetSolutionRuleset(binding, language);
+            if (!IsSolutionBound(binding, language, out var solutionRuleset, out var additionalFilePath))
+            {
+                return false;
+            }
+
+            var isAdditionalFileBound = ProjectHasAdditionalFile(project, additionalFilePath);
 
-            if (solutionRuleset == null)
+            if (!isAdditionalFileBound)
             {
                 return false;
             }
 
             var declarations = ruleSetInfoProvider.GetProjectRuleSetsDeclarations(project).ToArray();
 
-            return declarations.Length > 0 // Need at least one
-                   && declarations.All(declaration => IsRuleSetBound(project, declaration, solutionRuleset));
+            var isRuleSetBound = declarations.Length > 0 && declarations.All(declaration => IsRuleSetBound(project, declaration, solutionRuleset));
+
+            return isRuleSetBound;
+        }
+
+        private bool IsSolutionBound(BindingConfiguration binding, Language language, out RuleSet solutionRuleset, out string additionalFilePath)
+        {
+            solutionRuleset = null;
+            additionalFilePath = GetSolutionAdditionalFile(binding, language);
+
+            if (!fileSystem.File.Exists(additionalFilePath))
+            {
+                return false;
+            }
+
+            solutionRuleset = GetSolutionRuleset(binding, language);
+
+            return solutionRuleset != null;
+        }
+
+        private string GetSolutionAdditionalFile(BindingConfiguration binding, Language language)
```

Comment:
Exact duplication of the logic inside `CSharpVBBindingConfigProvider`.
Same goes for the ruleset filepath, that is also being duplicated here.

The reason for the duplication at the moment is because "IsBindingRequired" is called by `IUnboundProjectFinder`, which is called by `ErrorListInfoBarController`, at a stage where we didn't call yet `CSharpVBBindingConfigProvider`, so we don't have the `CSharpVBBindingConfig` with the file paths.
It's worth taking a look at `ErrorListInfoBarController` and see what we can do there -- if we can pass `CSharpVBBindingConfig` as part of IsBindingRequired, it would remove the duplication.

Reply:
i.e. the refactoring isn't complete yet.
Where/how is this infomration recorded? (needs to be recorded more permanently than in a review comment).

Reply:
As suggested above, the `solutionBindingFilePathGenerator` isn't required.

A less-ugly alternative to duplicating the code would be to make the path calculation methods `internal static` on the CSharpVB config provider and call them from here.
That way, at least the connection between the two bits of code is obvious.